### PR TITLE
Switch OTLP receiver to use GMP by default

### PR DESCRIPTION
**Pull request by quentinmit**
## Description
Add support for GMP exporting to the OTLP receiver, expose a `metrics_mode` config option to select it, and switch the default exporter from GCM to GMP.

## Related issue
b/274802213

## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [x] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [ ] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [ ] This PR introduces no new features.
  - [x] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

**Comment by quentinmit** [comment]
This PR is passing all tests and ready for review, but it's using an unreleased commit from operations-go, so I won't submit until that is released.

File: confgenerator/otel/processors.go [hunk 1]
```diff
@@ -336,3 +336,31 @@ func ModifyInstrumentationScope(name string, version string) Component {
...
+func GroupByGMPAttrs() Component {
+	return Component{
+		Type: "groupbyattrs",
+		Config: map[string]interface{}{
+			"keys": []string{"namespace", "cluster", "location"},
+		},
+	}
+}
+
+// GCPReosurceDetector returns a resourcedetection processor configured for only GCP.
```
Comment (jefferbrecht) [inline confgenerator/otel/processors.go:353]:
```suggestion
// GCPResourceDetector returns a resourcedetection processor configured for only GCP.
```

File: confgenerator/otel/processors.go [hunk 2]
```diff
@@ -336,3 +336,31 @@ func ModifyInstrumentationScope(name string, version string) Component {
...
+func GCPResourceDetector(override bool) Component {
+	config := map[string]interface{}{
+		"detectors": []string{"gcp"},
+	}
+	if override != true {
```
Comment (jefferbrecht) [inline confgenerator/otel/processors.go:358]:
```suggestion
	if !override {
```

File: confgenerator/otel/processors.go [hunk 3]
```diff
@@ -336,3 +336,31 @@ func ModifyInstrumentationScope(name string, version string) Component {
...
+
+// GroupByGMPAttrs moves the "namespace", "cluster", and "location"
+// metric attributes to resource attributes. The
+// googlemanagedprometheus exporter will use these resource attributes
+// to populate metric labels.
+func GroupByGMPAttrs() Component {
+	return Component{
+		Type: "groupbyattrs",
+		Config: map[string]interface{}{
+			"keys": []string{"namespace", "cluster", "location"},
```
Comment (jefferbrecht) [inline confgenerator/otel/processors.go:348]: Why aren't `instance_name` and `machine_type` included -- are they not resource attributes?

Comment (ridwanmsharif) [reply confgenerator/otel/processors.go:348]: Nope, they are just metric labels they are added for the prometheus receiver too so users can use those attributes in their queries. Full list of resource attributes are here: https://cloud.google.com/monitoring/api/resources#tag_prometheus_target

File: confgenerator/otel/modular.go [hunk 4]
```diff
@@ -53,7 +59,10 @@ type ReceiverPipeline struct {
 	// If a key is not in the map, the receiver pipeline will not be used for that pipeline type.
 	Processors map[string][]Component
 	// Type indicates if the pipeline outputs special metrics (either Prometheus or system metrics) that need to be handled with a special exporter.
-	Type ExporterType
+	ExporterTypes map[string]ExporterType
```
[the remark this answers is not in the document]
Comment (quentinmit) [reply confgenerator/otel/modular.go:62]: I guess we could take a dependency on OTel and use their enum: https://cs.github.com/open-telemetry/opentelemetry-collector/blob/508164044e55ea9e7f0b19acc824e095c89a5e19/component/config.go#L131

I'll punt that for now, though.

File: apps/otlp.go [hunk 5]
```diff
@@ -34,22 +34,104 @@ type ReceiverOTLP struct {
...
+						stmt(`attributes["location"]`, `resource.attributes["cloud.availability_zone"]`),
+						// namespace = host.id
+						stmt(`attributes["namespace"]`, `resource.attributes["host.id"]`),
+						// cluster = "__gce__"
+						stmt(`attributes["cluster"]`, `"__gce__"`),
+						// instance_name = host.name
+						stmt(`attributes["instance_name"]`, `resource.attributes["host.name"]`),
+						// machine_type = host.type
+						stmt(`attributes["machine_type"]`, `resource.attributes["host.type"]`),
+						// job and instance should be automatically set from service labels
```
[the remark this answers is not in the document]
Comment (quentinmit) [reply apps/otlp.go:70]: I am reusing the same integration test code as the prometheus receiver, so I believe these are already being tested. Perhaps the common code could be further enhanced to have better coverage.

File: confgenerator/otel/processors.go (hunk 2 above)
Comment (quentinmit) [reply confgenerator/otel/processors.go:358]: I wanted it like this to be more explicit about what we're checking.

**Review by jefferbrecht — APPROVED** [review]

**Pull request merged**